emscripten/ARToolKitJS.cpp

```diff
@@ -97,644 +98,763 @@ static int MARKER_INDEX_OUT_OF_BOUNDS = -3;
 
 static ARMarkerInfo gMarkerInfo;
 
-extern "C" {
+extern "C"
+{
 
-/**
-        NFT API bindings
-*/
+  /**
+          NFT API bindings
+  */
 
-void matrixLerp(ARdouble src[3][4], ARdouble dst[3][4],
-                float interpolationFactor) {
-  for (int i = 0; i < 3; i++) {
-    for (int j = 0; j < 4; j++) {
-      dst[i][j] = dst[i][j] + (src[i][j] - dst[i][j]) / interpolationFactor;
+  void matrixLerp(ARdouble src[3][4], ARdouble dst[3][4],
+                  float interpolationFactor)
+  {
+    for (int i = 0; i < 3; i++)
+    {
+      for (int j = 0; j < 4; j++)
+      {
+        dst[i][j] = dst[i][j] + (src[i][j] - dst[i][j]) / interpolationFactor;
+      }
     }
   }
-}
 
-emscripten::val getNFTMarkerInfo(int id, int markerIndex, emscripten::val videoFrame) {
-  if (arControllers.find(id) == arControllers.end()) {
-    return emscripten::val(ARCONTROLLER_NOT_FOUND);
-  }
-  arController *arc = &(arControllers[id]);
+  void passVideoData(int id, emscripten::val videoFrame, emscripten::val videoLuma)
```

Comment:
with this method we will pass data directly to the arController struct, so we avoid the multi copy operation.